File: Creational/Factory_Method/factory.py
from abc import ABC, abstractmethod

class Creator(ABC):
    @abstractmethod
    def factory_method(self):
        pass

    def some_operation(self) -> str:
        product = self.factory_method()
        return f"Creator: The same creator's code has worked with {product.operation()}"

class Product(ABC):
    @abstractmethod
    def operation(self) -> str:
        pass

class ConcreteProduct1(Product):
    def operation(self) -> str:
        return "{Result of Product 1}"

class ConcreteCreator1(Creator):
    def factory_method(self) -> Product:
        return ConcreteProduct1()

File: Creational/Factory_Method/factory.cs
using System;

public abstract class Creator {
    public abstract IProduct FactoryMethod();
    public string SomeOperation() {
        var product = FactoryMethod();
        return "Creator: The same creator's code has just worked with " + product.Operation();
    }
}

public interface IProduct {
    string Operation();
}

class ConcreteProduct1 : IProduct {
    public string Operation() => "{Result of Product 1}";
}

class ConcreteCreator1 : Creator {
    public override IProduct FactoryMethod() => new ConcreteProduct1();
}

// Usage
// var creator = new ConcreteCreator1();
// Console.WriteLine(creator.SomeOperation());